instrumentation/instacosmos/partition_key.go

```diff
@@ -0,0 +1,47 @@
+// (c) Copyright IBM Corp. 2024
+
+//go:build go1.18
+// +build go1.18
+
+package instacosmos
+
+import (
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
+)
+
+type pk struct {
+	value string
+}
+
+func NewPK(val string) *pk {
```

Comment:
Please add a one-line comment here. I think we can add comments for all exported (capitalized) names. It will be helpful for the documentation.

Also I have a suggestion using `NewPartitionKey` instead of `NewPK` for better clarity.
I think `instacosmos.NewPartitionKey` makes more sense and is more user-friendly.

Reply:
@Angith  :
Also, why can't we return the PartitionKey interface instead of *pk? I mean, NewPK is exported, but pk is a private struct. Whoever uses this externally won't be able to do anything with pk, right?

I think it makes more sense to return PartitionKey here. Feel free to correct me if I'm wrong.

Reply:
NewPK made it public only for internal integration tests. Since Cosmos Library denies access to the PartitionKey, we're also restricting access. But returning the interface is a good idea. Made the changes for it. Please review it.